### ISSUE TITLE
ESR Requirement: Auditing and Monitoring

### ISSUE THREAD

**Issue by destineeb** [commoncriteria/sdn-controller#27]
Auditing: The SDN controller must have maximum auditing and logging of all authorized API usage. 

Monitoring: The SDN must provide analytics on the API logs to detect API usage that does not comply with policies 

Auditing: The SDN controller must have maximum auditing and logging of all unauthorized API usage, especially API usage attempts from users that are not in the API roles. 

Can the below SFR cover the above three requirements? Refer to table 2 of  cPP_ND_v3.0e, 06-Dec-2023 55.

**FAU_GEN.1.1** The TSF shall be able to generate an audit record of the following auditable events: a. Start-up and shut-down of the audit functions; b. All auditable events for the not specified level of audit; and c. All administrative actions comprising:  

- Administrative login and logout (name of Administrator account shall be logged if individual accounts are required for Administrators).  

- Changes to TSF data related to configuration changes (in addition to the information that a change occurred it shall be logged what has been changed). cPP_ND_v3.0e, 06-Dec-2023 55 

- Generating/import of, changing, or deleting of cryptographic keys (in addition to the action itself a unique key name or key reference shall be logged).  

- Specifically defined auditable events listed in Table 2

**Comment by njgengo** [comment]
Yes, FAU_GEN.1.1 covers this SFR. We'll also add additional rows to Table 2 for additional auditable events for SDNCs.

**Comment by destineeb** [comment]
auditable events added in issue #33
